test/unit/limited_work_queue_test.rb

```diff
@@ -108,6 +108,26 @@ def merge(queue)
       assert_equal 1+2+4+32, c
     end
 
+    it "should ensure that the queue continues to function even when an exception is raised" do
+      mock(ExceptionHandling).log_error(anything, /LimitedWorkQueue encountered an exception/).twice
+      ExceptionalSynchrony::EMP.run_and_stop do
+        c = 0
+        last_start = nil; end_0 = nil; end_1 = nil
+        @queue.add! { c += 1; @em.sleep(0.001); end_0 = c+=1; raise "Boom" }
+        @queue.add! { c += 1; @em.sleep(0.001); end_1 = c+=1; raise "Boom" }
+        @queue.add! { last_start = c+= 1; @em.sleep(0.001); c+=1 }
+
+        3.times do
+          @em.sleep(0.005)
```

Comment:
This is pretty tight.  The math works, but if the system is slow for just a little bit it will cause a phantom.  Can you maybe change the 3 to a 30?